### ISSUE TITLE
How to save a MemNN model

### ISSUE THREAD

**Issue by chaitjo** [facebookresearch/ParlAI#253]
Hi! I was training a MemNN model on the Personalized Dialog dataset and wanted to save the model parameters with the highest validation score. I want to then use this model to obtain test set accuracy for both standard and OOV test sets.

I am able to run the training without any issues, but tried using the -mf flag to save the model. Here's the result-
```
(parlai) cil@FYP6:~/ParlAI/examples$ python memnn_luatorch_cpu/full_task_train.py -t personalized_dialog:SmallTask:1 -n
t 8 -mf /tmp/model_memnn
/home/cil/torch/install/bin/luajit
usage: full_task_train.py [-h] [-t TASK] [--download-path DOWNLOAD_PATH]
                          [-dt {train,train:ordered,valid,test}]
                          [-im IMAGE_MODE] [-nt NUMTHREADS] [-bs BATCHSIZE]
                          [-dp DATAPATH] [--dict-file DICT_FILE]
                          [--dict-initpath DICT_INITPATH]
                          [--dict-language DICT_LANGUAGE]
                          [--dict-max-ngram-size DICT_MAX_NGRAM_SIZE]
                          [--dict-minfreq DICT_MINFREQ]
                          [--dict-nulltoken DICT_NULLTOKEN]
                          [--dict-endtoken DICT_ENDTOKEN]
                          [--dict-unktoken DICT_UNKTOKEN]
                          [--dict-starttoken DICT_STARTTOKEN]
                          [--dict-maxexs DICT_MAXEXS] [--port PORT]
                          [--remote-address REMOTE_ADDRESS] [--remote-host]
                          [--remote-cmd REMOTE_CMD]
                          [--remote-args REMOTE_ARGS]
                          [--num-examples NUM_EXAMPLES] [--num-its NUM_ITS]
                          [--dict-max-exs DICT_MAX_EXS]
full_task_train.py: error: unrecognized arguments: -mf /tmp/model_memnn
```
How do I save and then restore a MemNN model with the framework?

**Comment by alexholdenmiller** [comment]
Unfortunately the lua memnn model uses it's own training loop which doesn't include this capacity yet. However, the new memnn model (in parlai/agents/memnn) does have this capacity--try this instead!

```bash
python train_model.py -t personalized_dialog:SmallTask:1 -bs 16 -m memnn -mf /tmp/model_memnn
```

**Comment by chaitjo** [comment]
Fantastic! Using the lua memnn was a bit confusing, it felt very undocumented so I had to feel my way around when figuring out hyperparameters or training settings.

**Comment by chaitjo** [comment]
The new model looks more straightforward to use. Will let you know if I face any issues!

**Comment by belbs** [comment]
Also, it might be good to use a larger batch size (maybe 64) in order to speed things up, since the candidate set is very large.
